Update to 2.2.0

soulard opened the pull request:
We update module to new version, with proper interface and new method for download products from Ecopresto.

soulard committed:
Modification du header licence, selon validator.prestashop.com

soulard committed:
Mise à jour version 2.2.0

soulard committed:
Mise à jour version 2.2.0

soulard committed:
Mise à jour header licence

soulard committed:
Gestion des erreurs SQL + Fonction update v220 manuelle

soulard committed:
Gestion des erreurs SQL + Fonction update v220 manuelle

soulard committed:
Correction dans la table

soulard committed:
Ajout fonction check doublon CSV

soulard committed:
Ajout fonction check doublon CSV

soulard committed:
Ajout fonction check doublon CSV

soulard committed:
Ajout fonction check doublon CSV

soulard committed:
Ajout de l'import par Parse CSV

soulard committed:
Ajout de l'import par Parse CSV

soulard committed:
Modification méthode parse

soulard committed:
Modification méthode parse

soulard committed:
Changement n° de version

soulard committed:
Suppression flushCatalogCSVMySQL

soulard committed:
Suppression flushCatalogCSVMySQL

soulard committed:
Suppression de 2 fonctions inutiles depuis le chg de methode parse

soulard committed:
Changement URL Documentation

soulard committed:
Ajout Readme

soulard committed:
Modification licence header

soulard committed:
Modification licence header

soulard committed:
Modification licence header

soulard committed:
Modification licence header

soulard committed:
Modifications suggérées par Prestashop Validator

soulard committed:
Modifications suggérées par Prestashop Validator

soulard committed:
Modifications suggérées par Prestashop Validator

soulard committed:
Modifications suggérées par Prestashop Validator

soulard committed:
Modifications suggérées par Prestashop Validator

soulard committed:
Modification traduction anglaise

soulard committed:
Modification traduction espagnole

soulard committed:
Ajout de la suppression du contenu des tables (au lieu de le faire depuis la classe)

soulard committed:
Suppression table catalogubrut du process d'installation

soulard committed:
Ajout d'instruction CREATE IF NOT EXIST

soulard committed:
Modification de la fonction setSupplier, si module déjà installé, on retrouve le supplier Ecopresto d'origine

soulard committed:
Suppression d'un appel à la méthode etatCatalogueBrut() qui générait une erreur SQL.

soulard committed:
Modification n° de version dans config.xml

soulard committed:
Modification du process multilangue : si la langue choisie n'est pas renseignée, on prend la version FR

Quetzacoalt91 commented on class/catalog.class.php:
```diff
@@ -836,7 +913,441 @@ public function GetFilecsv()
+	 */
+	public function GetCatalogCSV()
+	{
+		$domain = Configuration::get('PS_SHOP_DOMAIN');
```
This variable is unused

soulard replied:
We suppress it.

Quetzacoalt91 commented on class/catalog.class.php:
```diff
@@ -836,7 +913,441 @@ public function GetFilecsv()
+	{
+		$domain = Configuration::get('PS_SHOP_DOMAIN');
+		
+		$cle = $this->tabConfig['ID_ECOPRESTO'];
```
This variable is unused too

soulard replied:
We suppress it.

Quetzacoalt91 commented on class/catalog.class.php:
```diff
@@ -836,7 +913,441 @@ public function GetFilecsv()
+		
+		if ($download->load($filename) == true)
+		{
+			$download->saveTo(_PS_ROOT_DIR_.'/modules/ecopresto/files/'.$this->fichierImport);
```
This variable has the value _'catalogue.csv'_. It would be good to generate a name to avoid people accessing this file by writing the URL directly.

soulard replied:
We reuse old piece of code and don't change this part from previous version (even if some variable were use, it was always 'catalog.csv'. 
Currently, we are rewriting the generation process for this file from our server. 
I notice your remark and i'm agree with you, but we don't have time to finish it right now. So this part will be change in a future version.

Quetzacoalt91 commented on class/catalog.class.php:
```diff
@@ -836,7 +913,441 @@ public function GetFilecsv()
+	 * @param aucun
+	 * @return boolean false si erreur
+	 */
+	public function setCatalogCSVMySQL_Parsephp(&$tabErreur)
```
This function is unused

soulard replied:
We suppress it.

Quetzacoalt91 commented on ecopresto.php:
```diff
@@ -40,10 +48,10 @@ public function __construct()
 		$this->name = 'ecopresto';
 		$this->tab = 'shipping_logistics';
-		$this->version = '2.11';
+		$this->version = '2.2.0';
```
The new version is lower than **2.11**. You should write **2.20.0** instead

soulard replied:
We correct it.

Quetzacoalt91 commented on ecopresto.php:
```diff
@@ -119,7 +154,7 @@ private function controleLicence($id_eco, $typ)
 		else
 		{
 			if($id_eco != 'demo123456789demo123456789demo12')
-				Configuration::updateValue('ECOPRESTO_CONFIGURATION_OK', true);
```
Can you please keep this configuration data ? `ECOPRESTO_CONFIGURATION_OK` allow us to know when a module is properly configured.

soulard replied:
We re add it.

Quetzacoalt91 replied:
Thanks !

Quetzacoalt91 commented on ecopresto.php:
```diff
@@ -164,25 +199,178 @@ public function getContent()
+			}
+		}
+		/* Methode LOAD DATA
+		if (Tools::isSubmit('maj_catalogue_ecopresto'))
```
Can you please remove all these commented code ?

Quetzacoalt91 commented on sql/create.sql:
```diff
@@ -166,8 +166,6 @@ CREATE TABLE IF NOT EXISTS `PREFIX_ec_ecopresto_tracking` (
   PRIMARY KEY (`id`)
 ) ENGINE=MyISAM DEFAULT CHARSET=utf8;
-
-
```
A `CREATE TABLE` is missing here for `PREFIX_ec_ecopresto_cataloguebrut`

soulard replied:
PREFIX_ec_ecopresto_cataloguebrut is not need anymore (suppress function  setCatalogCSVMySQL_Parsephp which is not used anymore)

Quetzacoalt91 replied:
Okay, I was mistaken because of the file _sql/update-2.2.sql_

Quetzacoalt91 commented:
You should add a file *upgrade/Upgrade-2.20.0* to execute you file *sql/update-2.2.sql*

See this page for more information: http://doc.prestashop.com/display/PS15/Auto-updating+modules

soulard committed:
Changement n° de version en 2.20.0

soulard committed:
Deleting file

Version 2.20.0 doesn’t need change on db structure anymore.

soulard committed:
Those 2 vars aren't need anymore.

soulard committed:
Suppress setCatalogCSVMySQL_Parsephp

This function is not need anymore.

soulard committed:
allows Prestashop to know if all's right

soulard committed:
Remove commented code

soulard committed:
check file

Quetzacoalt91 commented:
Hi,

I've just installed the module and got the following notices. Did you test it with the _PS_MODE_DEV_ set true ?

![capture du 2015-03-31 14 13 26](https://cloud.githubusercontent.com/assets/6768917/6918594/6a10e1da-d7b0-11e4-9832-a24818de96a9.png)

soulard committed:
Add 3 lines to avoid Php notice in debug mode.

soulard commented:
We test module in debug mode and add some line to avoid PHP Notice.

soulard committed:
Add condition, print error only if mode debug is true

Quetzacoalt91 commented:
Hi @soulard,

Thanks for this first commit, but there are still notices on my settings page.
![capture du 2015-04-08 11 09 14](https://cloud.githubusercontent.com/assets/6768917/7042830/f20c79ca-dde4-11e4-8f1b-2fde80c3f30d.png)

I have also tried to use the producst import / export, and the popup never disappear (you can note the NaN% too)
![capture du 2015-04-08 11 09 38](https://cloud.githubusercontent.com/assets/6768917/7042896/9abf7efa-dde5-11e4-9f29-ba1182565297.png)

soulard committed:
Stock process optimisation

We add a test before loading product object. If a product is not in PS
Catalog, we don’t load it anymore.

soulard committed:
add to avoid PHP Notice in PS debug mode

soulard committed:
Suppress to avoid php notice in dev mode

soulard commented:
We commit this morning minor changes to avoid last messages in debug mode. We check all part of module and we don't see "PHP Notice" anymore.

We try to reproduce your popup error when syncing product. We only succeed one time: we mask all help message, we don't select any product in list, and then try to import (step 4). We don't make any change in this part of the process (step 3 and 4) so error should be the same in version 2.11.

Quetzacoalt91 commented:
Hi @soulard,

I thought you were interested to improve the module even if you did not have touched the concerned part, my bad.

Anyway, your PR has just been merged and the changes should be available soon.

Regards

soulard commented:
@Quetzacoalt91 : you're right: i try to improve the module even if code is not mine, but i have to choose what i can do in a limited time. Next, we had to create a cron job processing system, such as we can use in Drupal for example. Do you know if Prestashop as a such processing system ?

Quetzacoalt91 commented:
@soulard, we have a module called cronjobs which allow merchants to manage their cron tasks. Maybe it could suits your needs. The code is available [here](https://github.com/PrestaShop/cronjobs), and the module is installed by default on every new PrestaShop.
